torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1953,19 +1974,32 @@ def sample_inputs_hstack_dstack_vstack(op_info, device, dtype, requires_grad, **
         tensors = (make_arg(s1,), make_arg(s2,), make_arg(s3))
         yield SampleInput(tensors)
 
-def error_inputs_hstack_dstack_vstack(op, device):
+def _error_inputs_dstack_hstack_vstack_common(op, device, error_regex):
     make_arg = partial(make_tensor, dtype=torch.int32, device=device, requires_grad=False)
     tensor_shapes = (
         ((S,), (S, S, S, S), (S,)),
     )
     for s1, s2, s3 in tensor_shapes:
         tensors = (make_arg(s1,), make_arg(s2,), make_arg(s3))
         # Different dimension tensor
-        yield ErrorInput(SampleInput(tensors), error_regex="Tensors must have same number of dimensions")
+        yield ErrorInput(SampleInput(tensors), error_regex=error_regex)
 
     # empty tensor list
     yield ErrorInput(SampleInput(()), error_regex="expects a non-empty TensorList")
 
+def error_inputs_dstack(op, device):
+    yield from _error_inputs_dstack_hstack_vstack_common(
+        op=op, device=device, error_regex="Tensors must have same number of dimensions: got 3 and 4")
+
+def error_inputs_hstack(op, device):
+    yield from _error_inputs_dstack_hstack_vstack_common(
+        op=op, device=device, error_regex="Tensors must have same number of dimensions: got 1 and 4")
+
+def error_inputs_vstack(op, device):
+    yield from _error_inputs_dstack_hstack_vstack_common(
+        op=op, device=device, error_regex="Tensors must have same number of dimensions: got 2 and 4")
+
```

Comment:
Split these apart because I wanted to have precise error messages, but also didn't want to have a catch-all `else` branch.